numba/core/consts.py

```diff
@@ -93,8 +93,17 @@ def _infer_call(self, func, expr):
         _slice = func in (slice,)
         _exc = isinstance(func, type) and issubclass(func, BaseException)
         if _slice or _exc:
-            args = [self.infer_constant(a.name, loc=expr.loc) for a in
-                    expr.args]
+            args = []
+            for a in expr.args:
+                new_a = None
+                try:
+                    new_a = self.infer_constant(a.name, loc=expr.loc)
+                except ConstantInferenceError as cie:
```

Comment:
Is there any reason to always try infer constants when we have an exception? Is this to handle some of the existing unsupported cases?

If so I'd add a comment explaining that here and that we can still handle non-constant strings.

Reply:
> Is there any reason to always try infer constants when we have an exception?

I think this is for historic reasons as Numba can only handle exceptions is constant values, until this PR lands.